src/snowflake/snowpark/session.py

```diff
@@ -2430,6 +2430,10 @@ def _infer_is_return_table(
                     arg_types.append(convert_sp_to_sf_type(infer_type(arg)))
             func_signature = f"{sproc_name.upper()}({', '.join(arg_types)})"
 
+            if re.search("^\s*SYSTEM\$", func_signature):
```

Comment:
you are right. I didn't think we would allow users to create udf/sproc with name `system$`. I'll check if it is programmatically possible to determine this.

Reply:
right now, builtin stored procs include
- ASSOCIATE_SEMANTIC_CATEGORY_TAGS
- STORE_CLASSIFICATION
- SYSTEM$SEND_EMAIL
I'll add checks for these stored procs for now

Reply:
Feels like more stored procs under `GlobalServices/src/main/resources/com/snowflake/globaldecls` would qualify
i.e. `system$send_snowflake_notification`

Reply:
interesting. They don't show up in the following
```
show procedures;
select *  from table(result_scan(last_query_id())) where "is_builtin" ='Y';
```

even fewer system stored procs show up in prod deployment. Can you see them in preprod/prod?

Reply:
No I don't believe so